dawn/src/dawn/Optimizer/PassLocalVarType.cpp

```diff
@@ -0,0 +1,213 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "PassLocalVarType.h"
+#include "dawn/IIR/ASTExpr.h"
+#include "dawn/IIR/ASTStmt.h"
+#include "dawn/IIR/DoMethod.h"
+#include "dawn/IIR/StencilInstantiation.h"
+#include "dawn/IIR/StencilMetaInformation.h"
+
+namespace dawn {
+namespace {
+
+// Resets types of all variables to "not computed" (type_ = std::nullopt)
```

Comment:
I think I might prefer the metadata having a `resetVarTypes` function instead of allowing the whole world to arbitrarily change individual `LocalVariableData` information

Reply:
Ok.